website/blog/2021-07-28-release-apache-apisix-2.8-en.md

```diff
@@ -0,0 +1,143 @@
+### New plug-in: gzip
+
+Apache APISIX version 2.8 contains the gzip plug-in. Using the gzip plug-in, you can **dynamically set route-level gzip parameters**.
+
+An example of gzip configuration:
+
+```JSON
+{
+    "plugins": {
+        "gzip": {
+            "min_length": 20,
+            "http_version": 1.1,
+            "buffers": {
+                "number": 32,
+                "size": 4096
+            },
+            "types": [
+                "text/html"
+            ],
+            "comp_level": 1,
+            "vary": false
+        }
+    }
+}
+```
+
+### New plug-in: ua-restriction
+
+The `ua-restriction` plugin is used to check if the User-Agent is in the black and white list, which is a very common requirement and can be enabled by way of a plugin.
+
+An example of `ua-restriction` configuration:
+
+```JSON
+{
+    "plugins": {
+        "ua-restriction": {
+            "denylist": [
+                "my-bot1",
+                "(Baiduspider)/(\\d+)\\.(\\d+)"
+            ]
+        }
+    }
+}
+```
+
+### New way to develop: Support for executing specific logic by plug-ins
+
+Based on Apache APISIX architecture, many features are implemented by plug-ins. Starting from version 2.8, **Apache APISIX supports executing specific logic by plug-ins after selecting an upstream node.**
+
+Define the following method in the plug-in:
+
+```Lua
+function _M.balancer(conf, ctx)
+    core.log.notice("IP: ", ctx.balancer_ip, ", Port: ", ctx.balancer_port)
+end
+```
+
+In this example, the log will output the IP and Port of the upstream.
+
+**Which scenario will use the above method?**
+
+1. After selecting the upstream node and before accessing the upstream
+2. Before each retry
+
+For the best performance, the above method first runs in the access phase of OpenResty (APISIX actually selects the upstream node in the access phase) and the method does not overlap with the OpenResty phase of the same name.
+
+### New way to develop: Support for custom balancer
+
+In version 2.8, users can customize the balancer. **The balancer is loading with minimum number of connections, polling, consistency hash, etc.**
+
+Although Apache APISIX already provides a set of balancers, users may need to use balancers that are closely related to the business, such as: need to consider the server room, availability zone, etc. Supporting for custom balancer, users can develop their own balancer and load it via `require("apisix.balancer.your_balancer")`.
+
+Usually a custom balancer requires node to provide data which is from other than the host/port, you can put data in the metadata, for example:
+
+```JSON
+{
+    "nodes": [
+        { "host": "0.0.0.0", "port": 1980, "weight": 1, "metadata": {"b": 1} }
+    ]
+}
+```
+
+## Download
+
+Download Apache APISIX 2.8.0
+
+- Source code: please visit [download page](https://apisix.apache.org/downloads/)
+- Binary installation package: please visit [Installation Guide](https://apisix.apache.org/zh/docs/apisix/how-to-build/)
```

Comment:
I'd suggest  change `https://apisix.apache.org/zh/docs/apisix/how-to-build/` to `https://apisix.apache.org/docs/apisix/how-to-build/`, since this is an English doc, it should lead to another page in English as well.

Reply:
thanks